sol/main.c

```diff
@@ -575,6 +588,19 @@ sol_stage1(void *arg)
 	struct sol_config *sol_conf = arg;
 	int i;
 
+	static const struct rte_mbuf_dynfield priority_dynfield_desc = {
+		.name = "gatekeeper_dynfield_priority",
+		.size = sizeof(uint8_t),
+		.align = __alignof__(uint8_t),
+	};
+
+	sol_conf->priority_offset =
+		rte_mbuf_dynfield_register(&priority_dynfield_desc);
```

Comment:
The text In section "Detailed Description" of [this page](https://doc.dpdk.org/api/rte__mbuf__dyn_8h.html) of DPDK's documentation suggests that dynamic fields of `mbuf`'s are meant to **only** help with offloaded fields that a NIC supports. And `gatekeeper_dynfield_priority` is not supported for any NIC. :(

Notice that if a PMD (i.e. the NIC driver in DPDK) is not on board with the needed dynamic field, the PMD won't allocate space for this field when a packet is received.

The solution for us is to use the private area of `mbuf`'s, which Gatekeeper already uses. Search the code for `rte_mbuf_to_priv()` to find an entry point. And don't forget to update `rte_pktmbuf_pool_create_by_ops()` in `create_pktmbuf_pool()` to add space for the priority field.